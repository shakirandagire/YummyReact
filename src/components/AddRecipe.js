import React, { Component } from 'react';
import axios from 'axios';
import { notify } from 'react-notify-toast';
import Navigation from './Navigation';
import { BASE_URL } from '../../src/constants';

/**
 * Component for adding recipes
 */

class AddRecipe extends Component {
    state = {
      recipename: '',
      recipe_description: '',
      instructions: '',
    }

      handleInputChange = (event) => {
        const { name, value } = event.target;
        this.setState({ [name]: value });
      }
      // Method that handles adding of recipes

      handleAddRecipe = (event) => {
        const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
        const categoryId = this.props.match.params.id;
        const payload = {
          recipename: this.state.recipename,
          recipe_description: this.state.recipe_description,
          instructions: this.state.instructions,
        };
        event.preventDefault();
        axios
          .post(`${BASE_URL}/api/v1/categories/${categoryId}/recipes`, payload, { headers })
          .then((response) => {
            notify.show(response.data.message, 'success', 4000);
            this.props.history.push(`/view-recipes/${categoryId}/recipes`);
          })
          .catch((error) => {
            if (error.response) {
              notify.show(error.response.data.message);
            } else if (error.request) {
              notify.show("Request not made");
            }
          });
      }

      render() {
        return (
          <div className="my_container backgimg">
            <Navigation />

            <div className="mycontent">

              <form onSubmit={this.handleAddRecipe}>

                <div className="form-group">
                  Recipe Name:
                  <input type="text" name="recipename" className="form-control" required onChange={this.handleInputChange} value={this.state.recipename} />
                </div>
                <div className="form-group">
                  Description:
                  <input type="text-area" name="recipe_description" className="form-control" required onChange={this.handleInputChange} value={this.state.recipe_description} />
                </div>
                <div className="form-group">
                  Instructions:
                  <textarea name="instructions" className="form-control" required onChange={this.handleInputChange} value={this.state.instructions} />
                </div>

                <button type="submit" className="btn btn-danger formsave" >Save</button>
              </form>
            </div>

          </div>

        );
      }
}
export default AddRecipe;
