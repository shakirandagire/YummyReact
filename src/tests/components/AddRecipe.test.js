import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AddRecipe from '../../components/recipes/AddRecipe';

describe('<AddRecipe/>', () => {
  const params = {
    match: {
      params: {
        id: 1,
      },
    },
  };
  const component = shallow(<AddRecipe match={{ params }} />);
  const preventDefault = jest.fn();
  it('should render properly', () => {
    expect(component.length).toBe(1);
  });

  it('should render add recipes', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should have add recipe method', () => {
    expect(component.instance().handleAddRecipe({ preventDefault }));
  });
  it('should have initial state', () => {
    expect(component.state().recipename).toEqual("");
    expect(component.state().recipe_description).toEqual("");
    expect(component.state().instructions).toEqual("");
  });
  it('should change state', () => {
    component.setState({ recipename: 'brunch', recipe_description: 'breakfast and lunch',instructions: 'Sauages and eggs' });
    expect(component.find('[name="recipename"]').props().value).toEqual('brunch');
    expect(component.find('[name="recipe_description"]').props().value).toEqual('breakfast and lunch');
    expect(component.find('[name="instructions"]').props().value).toEqual('Sauages and eggs');
  });
  it('renders a form', () => {
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').simulate('submit', { preventDefault }));
    expect(preventDefault).toBeCalled();
  });
  it('should render form inputs', () => {
    expect(component.find('.recipename').length).toEqual(1);
    expect(component.find('.recipe_description').length).toEqual(1);
    expect(component.find('.instructions').length).toEqual(1);
  });
});
