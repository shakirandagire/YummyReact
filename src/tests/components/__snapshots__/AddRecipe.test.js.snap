// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRecipe/> should render add recipes 1`] = `
<div
  className="my_container backgimg"
>
  <Navigation />
  <div
    className="mycontent"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        Recipe Name:
        <input
          className="form-control recipename"
          name="recipename"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Description:
        <input
          className="form-control recipe_description"
          name="recipe_description"
          onChange={[Function]}
          required={true}
          type="text-area"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Instructions:
        <textarea
          className="form-control instructions"
          name="instructions"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <button
        className="btn btn-danger formsave"
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
